src/overlays/actors/ovl_En_Tite/z_en_tite.c

```diff
@@ -65,129 +112,1065 @@ static DamageTable D_80896AFC = {
+    } else if (this->unk_2BC > 0) {
+        this->unk_2BC--;
+        Math_StepToF(&this->actor.speedXZ, 10.0f, 0.3f);
+        this->actor.flags |= 0x1000000;
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        if (!func_80893A34(this, globalCtx)) {
+            this->unk_2BC = 0;
+        }
+    } else if ((this->unk_2BC == 0) && Math_StepToF(&this->actor.speedXZ, 0.0f, 0.3f)) {
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+        func_80893A18(this);
+        if ((Player_GetMask(globalCtx) == PLAYER_MASK_STONE_MASK) || (this->actor.xzDistToPlayer > 450.0f) || (this->actor.yDistToPlayer > 80.0f)) {
+            func_80893ED4(this);
+        } else if (!Actor_IsActorFacingLink(&this->actor, 0x2328)) {
+            func_808945EC(this);
+        } else {
+            func_80893A9C(this, globalCtx);
+        }
+    }
+    this->actor.shape.rot.y += (s16)(this->actor.speedXZ * 768.0f);
+}
 
-extern UNK_TYPE D_060004F8;
-extern UNK_TYPE D_0600069C;
-extern UNK_TYPE D_0600083C;
-extern UNK_TYPE D_06000A14;
-extern UNK_TYPE D_06000C70;
-extern UNK_TYPE D_060012E4;
+void func_8089595C(EnTite* this, GlobalContext* globalCtx) {
+    Vec3f sp2C;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tite/EnTite_Init.s")
+    sp2C.x = randPlusMinusPoint5Scaled(20.0f) + this->actor.world.pos.x;
+    sp2C.y = this->actor.world.pos.y + 15.0f;
+    sp2C.z = randPlusMinusPoint5Scaled(20.0f) + this->actor.world.pos.z;
+    func_800B0DE0(globalCtx, &sp2C, &D_801D15B0, &D_80896B44, &D_80896B3C, &D_80896B40, 500, 50);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tite/EnTite_Destroy.s")
+void func_80895A10(EnTite* this) {
+    s32 pad;
+    s16 rand;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tite/func_80893A18.s")
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000A14, 2.0f, 0.0f, 0.0f, 0, 4.0f);
+    this->actor.speedXZ = 0.0f;
+    rand = Rand_S16Offset(20, 20);
+    this->unk_2BC = ((Rand_ZeroOne() < 0.5f) ? -1 : 1) * rand;
+    this->actionFunc = func_80895AC0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tite/func_80893A34.s")
+void func_80895AC0(EnTite* this, GlobalContext* globalCtx) {
+    s16 temp_v1;
+    s32 temp_v0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tite/func_80893A9C.s")
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (func_801378B8(&this->skelAnime, 7.0f)) {
+        if (Rand_ZeroOne() < 0.25f) {
+            func_8089595C(this, globalCtx);
+        }
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_TEKU_WALK);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tite/func_80893ADC.s")
+    if ((this->unk_2BC == 1) || (this->unk_2BC == -1)) {
+        if (this->actor.bgCheckFlags & 8) {
+            temp_v1 = this->actor.shape.rot.y - this->actor.wallYaw;
+            temp_v0 = ABS_ALT(temp_v1);
+            if (temp_v0 > 0x3000) {
+                this->unk_2BC = (temp_v0 >> 9) * this->unk_2BC;
```

Comment:
```suggestion
                this->unk_2BC *= (temp_v0 >> 9);
```
This doesn't work does it?